### DISCUSSION TITLE
Quasar / Vite / TypeScript -> "module not found" in electron-preload.ts

### DISCUSSION THREAD

**Discussion by intelliBrain** [quasarframework/quasar#14201] [CLI - Electron mode] [answered]
I struggle with an annoying "module not found" error when using electron-preload.ts (TYPESCRIPT!) and trying to import external dependencies like @electron/remote for example.

the bug was already reported and partially discussed in discord "v2-electron": 
https://discordapp.com/channels/415874313728688138/856517669088002099/1008700437781823579 

how to reproduce it:

- git clone https://github.com/intelliBrain/quasar-electron-vite-ts-bug.git
- npm install
- quasar dev -m electron

NO-ERROR Scenario:
- make sure src-electron/electron-preload.ts [Line 17] is commented out like:
//const win = BrowserWindow.getFocusedWindow();

=> button clicks in the UI are successfully executed in electron-preload.ts (no error in console logs)
=> ![electron-preload-success](https://user-images.githubusercontent.com/5211988/184822209-ea80e4ee-ad8e-45fc-843c-9a4781f973da.png)

ERROR Scenario:
- uncomment src-electron/electron-preload.ts [Line 17] like:
const win = BrowserWindow.getFocusedWindow();

=> button clicks in the UI now results in "module not found" errors (see console logs in dev tools)
=> ![electron-preload-error](https://user-images.githubusercontent.com/5211988/184822231-efecd518-16a8-4d70-9653-021f029a3131.png)

Additional Info:
- as recommended in the quasar docs, npm packages used in electron-preload are installed not in "dev dependecies" but in "dependencies"
- discord user sherman claims that a similar configuration (quasar v2 / vite / but JAVASCRIPT instead of TYPESCRIPT) works without problems. However, I have not been able to verify this yet. 

NOTE: using JS instead of TS is not an option because of customer policies!

I need a quick fix / workaround ASAP to continue my work. Any input is highly appreciated!
Cheers

**Comment by HuntFeng** [comment, marked as answer]
I'm having the same issue here, I downgraded my electron from v20.x to v19.x, that solves my problem.

my quasar init config:

- Quasar CLI
- Quasar v2
- Javascript
- Vite
- using yarn

**Reply by intelliBrain** [reply]
@HuntFeng thank you so much for the tip! 

downgrading the electron version from v20.0.2 to v19.0.12 indeed did fix the issue!

my quasar init config:
- Quasar CLI
- Quasar v2.7.7 / Vue v3.2.37
- TypeScript
- Vite
- using npm

for reproduction see my repo: 
- https://github.com/intelliBrain/quasar-electron-vite-ts-bug.git
- branch name: "workaround_downgrade_electron_version"
